rspace/src/main/scala/coop/rchain/rspace/ReplayRSpace.scala

```diff
@@ -180,27 +175,25 @@ class ReplayRSpace[F[_]: Concurrent: ContextShift: Log: Metrics: Span, C, P, A,
       c =>
         store
           .getData(c)
-          .map(_.iterator.zipWithIndex.toSeq)
-          .map(
-            as => {
-              c -> {
-                if (c == channel)
-                  (Datum(data, persist, produceRef), -1) +: as
-                else as
-              }.filter {
-                matches(comm)
-              }
-            }
-          )
+          // Convert to immutable.Seq because of defined Traversable instance
+          .map(_.iterator.zipWithIndex.to[immutable.Seq])
+          .flatMap { datums =>
+            (if (c == channel) (Datum(data, persist, produceRef), -1) +: datums else datums)
+              .filterA(matches(comm))
+              .map(_.to[Seq])
+          }
+          .map((c, _))
```

Comment:
It is better to implement as `.map(c -> _)` to be consistent with `runMatcherConsume`.